### PR TITLE
Cleanup Standard Types Page

### PR DESCRIPTION
**Pull request by sydney-runkle**
## Change Summary

Cleanup standard types page (side menu, mostly) + add in suggestions from @Kludex from my last PR.

## Related issue number

Contributing to #7322 

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [x] Tests pass on CI
* [x] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: docs/api/standard_library_types.md [hunk 1]
```diff
@@ -168,10 +168,7 @@ Pydantic supports the following numeric types from the Python standard library:
 ### `enum.IntEnum`
 
 * Validation: Pydantic checks that the value is a valid `IntEnum` instance.
-
-#### subclass of `enum.IntEnum`
-
-* Validation: checks that the value is a valid member of the integer enum;
+* Validation for subclass of `enum.IntEnum`: checks that the value is a valid member of the integer enum;
```
Comment (hramezani) [inline docs/api/standard_library_types.md:171]:
```suggestion
* Validation for subclass of `enum.IntEnum`: checks that the value is a valid member of the integer enum.
```

**Review by hramezani — APPROVED** [review]

File: docs/api/standard_library_types.md (hunk 1 above)
Comment (Kludex) [reply docs/api/standard_library_types.md:171]: I think the `;` is correct, given there's the "see..." after.

File: docs/api/standard_library_types.md [hunk 2]
```diff
@@ -212,25 +209,12 @@ print(my_model.model_dump_json())  # (3)!
...
-Checks that the value is a valid member of the enum.
-
-### `enum.IntEnum`
-
-Checks that the value is a valid `IntEnum` instance.
-
-### Subclass of `enum.IntEnum`
-
-Checks that the value is a valid member of the integer enum.
+`enum.Enum` Checks that the value is a valid `Enum` instance.
```
Comment (Kludex) [inline docs/api/standard_library_types.md:216]:
```suggestion
`enum.Enum` checks that the value is a valid `Enum` instance.
```

File: docs/api/standard_library_types.md (hunk 1 above)
Comment (hramezani) [reply docs/api/standard_library_types.md:171]: Ok, I've deleted my comment

**Review by Kludex — APPROVED** [review]

**Pull request merged**